### PR TITLE
Copter: Classify failsafe action definitions

### PR DESCRIPTION
**Pull request by muramura**
I classify failsafe action definitions

### CODE REVIEW

**Comment by Pedals2Paddles** [comment]
Classy!  Nice thing about having all the failsafe actions in autotest, is we know this didn't break any functionality.  LGTM.

File: ArduCopter/Copter.h [hunk 1]
```diff
@@ -589,13 +589,13 @@ class Copter : public AP_Vehicle {
...
     };
 
-    enum Failsafe_Action {
-        Failsafe_Action_None           = 0,
-        Failsafe_Action_Land           = 1,
-        Failsafe_Action_RTL            = 2,
-        Failsafe_Action_SmartRTL       = 3,
-        Failsafe_Action_SmartRTL_Land  = 4,
-        Failsafe_Action_Terminate      = 5
+    enum class Failsafe_Action : int8_t {
```
Comment (peterbarker) [inline ArduCopter/Copter.h]: We've traditionally used upper-case for the actual enumeration class values.

It's not universal, but the vast majority follow that pattern.

Underscore should also be removed, so `FailsafeAction`, not `Failsafe_Action`.  You can browse through the code and find that that too is prevalent.

See `FailsafeOption` just down below, for example!

Also, this should be `uint8_t`, not `int8_t`.

File: ArduCopter/Copter.h [hunk 2]
```diff
@@ -607,17 +607,17 @@ class Copter : public AP_Vehicle {
     };
 
     static constexpr int8_t _failsafe_priorities[] = {
-                                                      Failsafe_Action_Terminate,
-                                                      Failsafe_Action_Land,
-                                                      Failsafe_Action_RTL,
-                                                      Failsafe_Action_SmartRTL_Land,
-                                                      Failsafe_Action_SmartRTL,
-                                                      Failsafe_Action_None,
+                                                      static_cast<int8_t>(Failsafe_Action::Terminate),
```
Comment (peterbarker) [inline ArduCopter/Copter.h]: I wonder if we should take this opportunity to have these indented further to the left?

Comment (WickedShell) [reply ArduCopter/Copter.h]: No. It's nicely aligned on it's parent bracket :)

Comment (peterbarker) [reply ArduCopter/Copter.h]: image

There we go - immortalised :-P

**Comment by khancyr** [comment]
@muramura hello, could you do the changes ask by Peter ? thanks !

**Pull request closed without merging**